commit 64a7238c3c082b174e47c3896d6eb0f5c52ab461
Author: Albert Astals Cid <aacid@kde.org>
Date:   Sat Nov 30 22:33:39 2013 +0100

    Make KLimitedIODevice::bytesAvailable return the numbers of bytesAvaliable to read
    
    At the moment KLimitedIODevice::bytesAvailable is returning at least the size of the file, so QIODevice::atEnd never returns true because it thinks there are more bytes to read. This makes that if you feed an "invalid" svg like "<?xml version="1.0"?>" to the QImageReader it infinite loops believing there will be more stuff to read but then read() always returns 0 but then bytesAvailable says there are more and it stays there forever.
    
    BUGS: 328182
    FIXED-IN: 4.11.5
    REVIEW: 114226

diff --git a/kdecore/io/klimitediodevice.cpp b/kdecore/io/klimitediodevice.cpp
index c93463b..9ec5901 100644
--- a/kdecore/io/klimitediodevice.cpp
+++ b/kdecore/io/klimitediodevice.cpp
@@ -71,7 +71,7 @@ bool KLimitedIODevice::seek( qint64 pos )
 
 qint64 KLimitedIODevice::bytesAvailable() const
 {
-     return m_length + QIODevice::bytesAvailable();
+     return QIODevice::bytesAvailable();
 }
 
 bool KLimitedIODevice::isSequential() const
